tests/test_main.py

```diff
@@ -0,0 +1,369 @@
+import pytest
+import datetime
+
+from django.contrib.auth import get_user_model
+from django.conf import settings
+from django.db.utils import IntegrityError
+from django.core.exceptions import ValidationError
+
+from db.models import (
+    Actor,
+    Genre,
+    Movie,
+    MovieSession,
+    CinemaHall,
+    Order,
+    Ticket
+)
+from services.movie_session import (
+    get_tickets_of_taken_seats
+)
+from services.user import create_user, get_user, update_user
+from services.order import create_order, get_orders
+
+
+@pytest.fixture()
+def database_data():
+    action = Genre.objects.create(name="Action")
+    drama = Genre.objects.create(name="Drama")
+    western = Genre.objects.create(name="Western")
+
+    reeves = Actor.objects.create(first_name="Keanu", last_name="Reeves")
+    johansson = Actor.objects.create(first_name="Scarlett",
+                                     last_name="Johansson")
+    clooney = Actor.objects.create(first_name="George", last_name="Clooney")
+
+    matrix = Movie.objects.create(title="Matrix", description="Matrix movie")
+    matrix.actors.add(reeves)
+    matrix.actors.add(johansson)
+    matrix.genres.add(action)
+
+    batman = Movie.objects.create(title="Batman", description="Batman movie")
+    batman.genres.add(drama)
+    batman.actors.add(clooney)
+
+    titanic = Movie.objects.create(title="Titanic",
+                                   description="Titanic movie")
+    titanic.genres.add(drama, action)
+
+    good_bad = Movie.objects.create(
+        title="The Good, the Bad and the Ugly",
+        description="The Good, the Bad and the Ugly movie",
+    )
+    good_bad.genres.add(western)
+
+    blue = CinemaHall.objects.create(name="Blue", rows=10, seats_in_row=12)
+    vip = CinemaHall.objects.create(name="VIP", rows=4, seats_in_row=6)
+    cheap = CinemaHall.objects.create(name="Cheap", rows=15, seats_in_row=27)
+
+    MovieSession.objects.create(
+        show_time=datetime.datetime(2019, 8, 19, 20, 30),
+        cinema_hall=blue,
+        movie=matrix
+    )
+    MovieSession.objects.create(
+        show_time=datetime.datetime(2017, 8, 19, 11, 10),
+        cinema_hall=cheap,
+        movie=titanic,
+    )
+    MovieSession.objects.create(
+        show_time=datetime.datetime(2021, 4, 3, 13, 50),
+        cinema_hall=vip,
+        movie=good_bad
+    )
+
+    MovieSession.objects.create(
+        show_time=datetime.datetime(2021, 4, 3, 16, 30),
+        cinema_hall=cheap,
+        movie=matrix
+    )
+
+    get_user_model().objects.create_user(username="user1",
+                                         password="pass1234")
+    get_user_model().objects.create_user(username="user2",
+                                         password="pass1234")
+
+    Order.objects.create(user_id=1)
```

Comment:
This is a bit tricky to depend on these `id` - you'd better use here users, that you created earlier:
```python
# btw - it is better to split it to different fixture (and return these 2 users in order)
user1 = get_user_model().objects.create_user(username="user1",
                                         password="pass1234")

rder.objects.create(user=user1)
```